packages/block-editor/src/components/url-input/index.js

```diff
@@ -82,6 +84,11 @@ class URLInput extends Component {
 		if ( this.shouldShowInitialSuggestions() ) {
 			this.updateSuggestions();
 		}
+
+		// Reset the component to the starting state when things are cleared
+		if ( prevProps.value && ! value ) {
```

Comment:
If you're doing this only to reset the input when the "RESET" button is clicked in LinkControl then you _could_ consider passing a ref into `URLInput` and then when `onReset` is called in `LinkControl` you could `.focus()` on the ref which would then be forwarded down to the `<input>` within `URLInput`.

Reply:
It would be ideal to be able to run this reset only when the rest button is clicked over in `packages/block-editor/src/components/link-control/search-input.js`, but I wasn't able to get something there to work due to the lack of input reference. We may be over-engineering things if we use a forward ref though.

Reply:
Looks like reset was removed by @aduth so we can ditch this.